Makefile

```diff
@@ -111,37 +115,49 @@ export STATUSGO
 export STATUSGO_LIBDIR
 export LIBSTATUS_EXT
 
+KEYCARDGO := vendor/nim-keycard-go/go/keycard/build/libkeycard/libkeycard.$(LIBKEYCARD_EXT)
+KEYCARDGO_LIBDIR := $(shell pwd)/$(shell dirname "$(KEYCARDGO)")
+export KEYCARDGO
+export KEYCARDGO_LIBDIR
+export LIBKEYCARD_EXT
+
 status-go: $(STATUSGO)
 $(STATUSGO): | deps
 	echo -e $(BUILD_MSG) "status-go"
 	+ cd vendor/status-go && \
 	  $(MAKE) statusgo-shared-library $(HANDLE_OUTPUT)
 
+keycard-go: $(KEYCARDGO)
+$(KEYCARDGO): | deps
+	echo -e $(BUILD_MSG) "keycard-go"
+	+ cd vendor/nim-keycard-go && \
+	  $(MAKE) build-keycard-go $(HANDLE_OUTPUT)
+
 LIBSTATUSLIB := build/$@.$(LIBSTATUS_EXT).0
-libstatuslib: | $(STATUSGO)
+libstatuslib: | $(STATUSGO) $(KEYCARDGO)
 	echo -e $(BUILD_MSG) "$@" && \
-		$(ENV_SCRIPT) nim c $(NIM_PARAMS) $(NIM_EXTRA_PARAMS) --passL:"-L$(STATUSGO_LIBDIR)" --passL:"-lstatus" -o:build/$@.$(LIBSTATUS_EXT).0 -d:ssl --app:lib --noMain --header --nimcache:nimcache/libstatuslib statuslib.nim && \
+		$(ENV_SCRIPT) nim c $(NIM_PARAMS) $(NIM_EXTRA_PARAMS) --passL:"-L$(STATUSGO_LIBDIR)" --passL:"-lstatus" --passL:"-L$(KEYCARDGO_LIBDIR)" --passL="-lkeycard" -o:build/$@.$(LIBSTATUS_EXT).0 -d:ssl --app:lib --noMain --header --nimcache:nimcache/libstatuslib statuslib.nim && \
 		rm -f build/$@.$(LIBSTATUS_EXT) && \
 		ln -s $@.$(LIBSTATUS_EXT).0 build/$@.so && \
 		cp nimcache/libstatuslib/*.h build/. && \
 		[[ $$? = 0 ]]
 
 # libraries for dynamic linking of non-Nim objects
-EXTRA_LIBS_DYNAMIC := -L"$(CURDIR)/build" -lstatuslib -lm -L"$(STATUSGO_LIBDIR)" -lstatus
+EXTRA_LIBS_DYNAMIC := -L"$(CURDIR)/build" -lstatuslib -lm -L"$(STATUSGO_LIBDIR)" -lstatus -L"$(KEYCARDGO_LIBDIR)" -lkeycard
 build_ctest: | $(LIBSTATUSLIB) build deps
 	echo -e $(BUILD_MSG) "build/ctest" && \
 		 $(CC) test/main.c -Wl,-rpath,'$$ORIGIN' -I./vendor/nimbus-build-system/vendor/Nim/lib $(EXTRA_LIBS_DYNAMIC) -g -o build/ctest
 
 LD_LIBRARY_PATH_NIMBLE := $${LD_LIBRARY_PATH}
 ifneq ($(detected_OS),Windows)
  ifneq ($(detected_OS),Darwin)
-  LD_LIBRARY_PATH_NIMBLE := $(STATUSGO_LIBDIR):$(LD_LIBRARY_PATH_NIMBLE)
+	LD_LIBRARY_PATH_NIMBLE := $(STATUSGO_LIBDIR):$(KEYCARDGO_LIBDIR):$(LD_LIBRARY_PATH_NIMBLE)
```

Comment:
My understanding is that it's best to not use tabs in non-recipe sections of a Makefile, though that understanding could be outdated or I misunderstood the guidance re: when to use and not use tabs.